src/thunderbots/software/ai/passing/evaluation.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Implementation of evaluation functions for passing
+ */
+
+// TODO: declare things const& and whatnot to maybe improve speed
+
+#include "ai/passing/evaluation.h"
+
+#include "util/parameter/dynamic_parameters.h"
+
+using namespace AI::Passing;
+
+double AI::Passing::getStaticPositionQuality(const Field& field, const Point& position)
+{
+    // This constant is used to determine how steep the sigmoid slopes below are
+    static const double sigmoid_width = 0.1;
+
+    // The offset from the sides of the field for the center of the sigmoid functions
+    double x_offset =
+        Util::DynamicParameters::AI::Passing::static_position_quality_x_offset.value();
+    double y_offset =
+        Util::DynamicParameters::AI::Passing::static_position_quality_y_offset.value();
+    double friendly_goal_weight =
+        Util::DynamicParameters::AI::Passing::
+            static_position_quality_friendly_goal_distance_weight.value();
+
+    // Make a slightly smaller field, and positive weight values in this reduced field
+    double half_field_length = field.length() / 2;
+    double half_field_width  = field.width() / 2;
+    Rectangle reduced_size_field(
+        Point(-half_field_length + x_offset, -half_field_width + y_offset),
+        Point(half_field_length - x_offset, half_field_width - y_offset));
+    double on_field_quality =
+        rectangleSigmoid(reduced_size_field, position, sigmoid_width);
+
+    // Add a negative weight for positions closer to our goal
+    Vector vec_to_friendly_goal      = Vector(field.friendlyGoal().x() - position.x(),
+                                         field.friendlyGoal().y() - position.y());
+    double distance_to_friendly_goal = vec_to_friendly_goal.len();
+    double near_friendly_goal_quality =
+        (1 -
+         std::exp(-friendly_goal_weight * (std::pow(5, -2 + distance_to_friendly_goal))));
```

Comment:
The `5` and `-2` seem like magic numbers here. Can we add a comment or replace them with descriptive variables?

Reply:
Hmm, not really. This was pretty much a "gave me the curve I was looking for" sort of thing

Reply:
I could add a note here indicating as such?

Reply:
After a re-read I'm fine with this. Can we use the `sigmoid` function here though?

Reply:
Nope, see here to see what it looks like:https://www.desmos.com/calculator/enb3ydfccw